### PR TITLE
AO3-5046 AO3-5854 Modify OrphansController to improve handling of pseud orphaning.

### PR DESCRIPTION
**Pull request by tickinginstant**
## Issue

https://otwarchive.atlassian.net/browse/AO3-5046
https://otwarchive.atlassian.net/browse/AO3-5854

## Purpose

- Rewrite `OrphansController#new` and `OrphansHelper#render_orphan_partial` so that the unused partials `orphans/_orphan_pseud` and `orphans/_orphan_user` are used.
- Rewrite the newly used partials so that they display a list of works to be orphaned, and have caution text echoing the existing text in `orphans/_orphan_work` (with a few slight changes).
- Rewrite `OrphansController#create` so that the `pseud_id` can be passed as an optional argument, to orphan only a single pseud (and make it so that only that pseud is removed, even if the user has another pseud on the same work).
- Add a new cucumber test for the changed behavior when orphaning a pseud.
- Add a collection of new RSpec tests for `OrphansController#new` and `OrphansController#create`.

## Testing Instructions

See JIRA for instructions for AO3-5046, and see the wiki for instructions on how to test AO3-5854.

### CODE REVIEW

File: app/controllers/orphans_controller.rb [hunk 1]
```diff
@@ -2,54 +2,88 @@ class OrphansController < ApplicationController
...
-  def create
-    new_orphans = {}
-    use_default = params[:use_default] == "true"
-    if !@pseuds.blank? && Creatorship.orphan(@pseuds, @orphans, use_default)
-      flash[:notice] = ts('Orphaning was successful.')
-      redirect_to(current_user)
+  # Given an ActiveRecord item and an ActiveRecord relation, check whether the
+  # item is in the relation. If not, show a flash error.
+  def check_one_owned(chosen_item, all_owned_items)
+    show_orphan_permission_error unless all_owned_items.where(id: chosen_item.id).exists?
```
[the remark this answers is not in the document]
Comment (tickinginstant) [reply app/controllers/orphans_controller.rb]: Good call! Looks like that would work fine.

File: app/controllers/orphans_controller.rb [hunk 2]
```diff
@@ -2,54 +2,88 @@ class OrphansController < ApplicationController
...
+  end
+
+  # Given a collection of ActiveRecords and an ActiveRecord relation, check
+  # whether all items in the collection are contained in the relation. If not,
+  # show a flash error.
+  def check_all_owned(chosen_items, all_owned_items)
+    chosen_ids = chosen_items.map(&:id)
+    owned_ids = all_owned_items.where(id: chosen_ids).pluck(:id)
+    unowned_ids = chosen_ids - owned_ids
```
[the remark this answers is not in the document]
Comment (tickinginstant) [reply app/controllers/orphans_controller.rb]: I don't think so, because that would return the set of all IDs in `all_owned_items` that aren't in `chosen_items`.  The `unowned_ids` array is supposed to be the reverse: all IDs in `chosen_items` that aren't in `all_owned_items`.

(I could probably do the reverse by making sure that `chosen_items` is a relation, and plucking all IDs from `all_owned_items`. But I was trying to reduce the number of queries.)

[the remark this answers is not in the document]
Comment (redsummernight) [reply app/controllers/orphans_controller.rb]: I missed that, this is good as is.

File: app/views/orphans/_orphan_pseud.html.erb [hunk 3]
```diff
@@ -1,22 +1,43 @@
...
+<h2 class="heading"><%= ts("Orphan All Works by %{name}", name: pseud.name) %></h2>
 
 <p class="caution">
-<%= ts("Are you sure you want to") %>
-<strong><%= ts("permanently") %></strong>
-<%= ts("remove all identifying information from all the works you've posted to 
-the archive under this pseud, along with their chapters, 
-associated series and any feedback replies you may have left?") %>
+  <%= ts("Orphaning all works by %{name} will", name: pseud.name)%>
+  <strong><%= ts("permanently")%></strong>
```
[the remark this answers is not in the document]
Comment (sarken) [reply app/views/orphans/_orphan_pseud.html.erb]: Yeah, we definitely want to move toward all sentences being a single translatable string at some point.

File: app/views/orphans/_orphan_pseud.html.erb [hunk 4]
```diff
@@ -1,22 +1,43 @@
...
+  <%= ts("detach the pseud %{name} from the following work(s), their chapters, associated series, and any feedback replies you may have left on them.", name: pseud.name) %>
```
Comment (sarken) [inline app/views/orphans/_orphan_pseud.html.erb]: Why the change from "remove" to "detach"? (I'm thinking "remove" is easier for people who don't speak English as a first language.)

Comment (tickinginstant) [reply app/views/orphans/_orphan_pseud.html.erb]: I merged in a lot of the wording from the `orphans/_orphan_work` partial, which uses detach:
https://github.com/otwcode/otwarchive/blob/156720eb90c391d183f598edf531f940b0fd5fe3/app/views/orphans/_orphan_work.html.erb#L4-L8
Do you want me to change it to remove everywhere?

(Also, what about information vs. data? The work partial uses "data"; the user partial uses "information".)

Comment (sarken) [reply app/views/orphans/_orphan_pseud.html.erb]: "Remove" everywhere would be great, thanks! And "data" should be good for the other one. (I ran the sentence through the [Readability Analyzer](https://datayze.com/readability-analyzer.php) our Docs team uses and it had a surprisingly higher level of reading ease with "data" than "information." I wasn't expecting that.)

Comment (tickinginstant) [reply app/views/orphans/_orphan_pseud.html.erb]: Remove and data it is!

(It may be related to the number of syllables, rather than the frequency of the word.)
